wayrs-client/src/connection.rs

```diff
@@ -37,14 +36,16 @@ pub enum ConnectError {
 /// and dispatches object events.
 ///
 /// Set `WAYLAND_DEBUG=1` environment variable to get debug messages.
-pub struct Connection<D> {
+pub struct Connection<D, T: ClientTransport = UnixStream> {
     #[cfg(feature = "tokio")]
     async_fd: Option<AsyncFd<RawFd>>,
+    #[cfg(feature = "tokio")]
+    read_async_fd: Option<AsyncFd<RawFd>>,
 
-    socket: BufferedSocket<UnixStream>,
+    pub socket: BufferedSocket<T>,
```

Comment:
If you made the socket public to have access to the transport T, a pair of `fn  transport(&self) -> &T` and `fn transport_mut(&mut self) -> &mut T` would be better.